mantidproject/mantid
# Reflectometry GUI Restructuring 16: stitch parameters

***PR ~~#25921~~ should be merged first.*** In the meantime this can be reviewed by comparing against [that branch](https://github.com/gemmaguest/mantid/compare/23027_refl_gui_disable_editing_keys...23027_refl_gui_stitch_params)

**Description of work.**

Ongoing work to integrate the `JobTreeView` into the `ISIS Reflectometry` GUI. This PR is being merged to a feature branch because not all functionality is implemented yet.

This PR adds functionality to take the bin width for the stitching params from the dQ/Q resolution if the stitching params are not explicitly set. The order of precedence is:
 - The params in the `Stitch1DMany` box on the **Experiment** tab (e.g. `Params=0.1`)
 - `dQ/Q` from the table on the **Runs** tab (negated to give log binning), for the first valid row for the group being stitched.
 - `dQ/Q` (again negated) from the Per-angle defaults box on the **Experiment** tab for a wildcard row (that is, a row without an angle set). In reality the code for this last option will never be called because it will be populated in the `dQ/Q` outputs in the runs table but the code is included for completeness and future-proofing.

**To test:**

- Open the `ISIS Reflectometry` interface
- Add a group with two rows, with runs `13460`, `13461` and angles `0.5` and `2.3` respectively and click Process.
- The output `dQ/Q` is updated in both rows. The result is `0.047641` for the first row.
- Check the history for `IvsQ_13460_13461`. The params for `Stitch1DMany` should be taken from `dQ/Q`, i.e. `-0.047641`.
- Change the `dQ/Q` in the per-angle defaults box on the **Experiment** tab to `0.01` and re-process. The Params should now be `-0.01`. `dQ/Q` on the **Runs** tab is also updated to `0.01` for both rows.
- Set `dQ/Q` on the **Runs** tab for the **second** row to `0.02` and re-process. The Params are unchanged at `-0.01`.
- Set the `dQ/Q` on the **Runs** tab for the **first** row to `0.03`. The Params are updated to `-0.03`.
- Set the `Stitch1DMany` entry on the **Experiment** tab to `StartOverlaps=0.03,EndOverlaps=0.07`. Check that these properties were used in the history but that the Params used remains at `-0.03`.
- Set the `Stitch1DMany` entry on the **Experiment** tab to `Params=0.05` and reprocess. The Params are updated to `0.05`.

Refs #23027 

*This does not require release notes* because **it re-implements functionality in the original GUI**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
